budgetportal/models.py

```diff
@@ -1412,9 +1417,178 @@ class PackageWithoutGroupException(Exception):
     pass
 
 
+class InfrastructureProject:
+    """ Represents an infrastructure project stored in CKAN """
+
+    def __init__(self, **kwargs):
+        self.records = kwargs.get('records')
+        self.name = self.records[0]['Project name']
+        self.stage = self.records[0]['Current project stage']
+        self.department_name = self.records[0]['Department']
+        self.description = self.records[0]['Project description']
+        self.total_budget = float(self.records[0]['Total project cost'])
+        self.nature_of_investment = self.records[0]['Nature of investment']
+        self.infrastructure_type = self.records[0]['Infrastructure type']
+        self.raw_coordinate_string = self.records[0]['GPS code']
+        self.complete_expenditure = self._build_complete_expenditure(self.records)
+        self.cleaned_coordinates = self._clean_coordinates(self.raw_coordinate_string)
+        self.projected_expenditure = self._calculate_projected_expenditure(self.records)
+
+    @classmethod
+    def get_dataset(cls):
+        """ Return the first dataset in the Infrastructure Projects group. """
+        query = {
+            'q': '',
+            'fq': ('+organization:"national-treasury"'
+                   '+vocab_spheres:"national"'
+                   '+groups:"infrastructure-projects"'),
+            'rows': 1,
+        }
+        response = ckan.action.package_search(**query)
+        logger.info(
+            "query %s\nreturned %d results",
+            pformat(query),
+            len(response['results'])
+        )
+        if response['results']:
+            return Dataset.from_package(response['results'][0])
+        else:
+            raise Http404()
```

Comment:
This works, although it'd be better if models.py indicated that it's missing to the view code, and the view code itself raised the 404, since a 404 is very much an http thing which is the view's concern, while the model is concerned with data and not the protocol used to access it.